chemprop/uncertainty/evaluator.py

```diff
@@ -1,19 +1,40 @@
-from abc import abstractmethod
+from abc import ABC, abstractmethod
 
 from torch import Tensor
 
 from chemprop.utils.registry import ClassRegistry
 
 
-class UncertaintyEvaluator:
-    """
-    A class for evaluating the effectiveness of uncertainty estimates with metrics.
-    """
+class UncertaintyEvaluator(ABC):
+    """A :class:`UncertaintyEvaluator` evaluates the quality of uncertainty estimates using a specified metric."""
 
     @abstractmethod
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        """
-        Evaluate the performance of uncertainty predictions against the model target values.
+        """Evaluate the performance of uncertainty predictions against the model target values.
+
+        Parameters
+        ----------
+        preds: Tensor
+            the predictions for regression tasks. It is a tensor of the shape of ``n x t``, where ``n`` is the number of input
+            molecules/reactions, and ``t`` is the number of tasks.
+        uncs: Tensor
+            the predicted uncertainties of varying shape depending on the task type:
+
+            * regression/binary classification: ``n x t``
+
+            * multiclass classification: ``n x t x c``, where ``c`` is the number of classes
+            .. note::
+                The `preds` is only needed for regression tasks. The `uncs` would be the predicted variance for regression tasks,
+                the predicted probability of class 1 for binary classification and the probabilities for each class for multiclass classification.
+        targets: Tensor
+            a tensor of the shape ``n x t``
+        mask: Tensor
+            a tensor of the shape ``n x t`` indicating whether the given values should be used in the fitting
+
+        Returns
+        -------
+        Tensor
+            a tensor of the shape ``t`` containing the evaluated metrics
```

Comment:
Suggestion on consistency:
you can write the shape as ``t x 1`` instead of just ``t``.